packages/host-portal-app/src/components/remote-view.tsx

```diff
@@ -10,55 +9,69 @@ import React, {
 import { Manifest, View } from '../types';
 
 const loading = Symbol('loading');
-type MicroFrontendState = Record<string, React.ComponentType | typeof loading>;
 
-const views = createContext<
-  [
+type MicroFrontendState = Record<string, React.ComponentType | typeof loading>;
+type GlobalStylesheetSet = Set<string>;
+type RemoteViewsContext = {
+  microFrontends: [
     MicroFrontendState,
     (value: (prevState: MicroFrontendState) => MicroFrontendState) => void,
-  ]
->([{}, () => null]);
-
-async function loadRemoteView(baseUrl: string): Promise<ComponentType | void> {
-  const response = await fetch(`${baseUrl}/package.json`);
-  const manifest = (await response.json()) as Manifest;
-
-  // TODO: check for (and eventually load) global CSS
-
-  // Load microfrontend's local style
-  if (manifest.style) {
-    document.head.insertAdjacentHTML(
-      'beforeend',
-      `<link rel='stylesheet' href='${baseUrl}${manifest.style}' />`,
-    );
-  }
-
-  // Dynamically import ESM entrypoint
-  if (manifest.module) {
-    const { default: LoadedView } = (await import(
-      /* webpackIgnore: true */ `${baseUrl}${manifest.module}`
-    )) as View;
+  ];
+  globalStyles: [
+    GlobalStylesheetSet,
+    (value: (prevState: GlobalStylesheetSet) => GlobalStylesheetSet) => void,
+  ];
+};
 
-    return LoadedView;
-  }
-}
+export const viewContext = createContext<RemoteViewsContext>(
+  {} as RemoteViewsContext,
```

Comment:
This isn't a valid default implementation of a `RemoteViewsContext`, using a `RemoteView` without a containing `RemoteViews` will throw when the `useRemoteView` attempts to destructure the context data.

Reply:
n/a anymore, see https://github.com/cristiano-belloni/modular-microfrontends/pull/7#discussion_r1014803846